app/src/main/java/com/futurice/freesound/feature/common/ui/adapter/IListItemComparator.java

```diff
@@ -0,0 +1,26 @@
+package com.futurice.freesound.feature.common.ui.adapter;
+
+import com.futurice.freesound.feature.common.DisplayableItem;
+
+public interface IListItemComparator {
+
+    /**
+     * Decides whether two {@link DisplayableItem} represent the same Item.
+     * For example, if your items have unique ids, this method should check their id equality.
+     *
+     * @return True if the two items represent the same object or false if they are different.
+     */
+    boolean areItemsTheSame(final DisplayableItem item1, final DisplayableItem item2);
+
+    /**
+     * Checks whether the visual representation of two {@link DisplayableItem}s is the same.
+     *
+     * This method is called only if {@link #areItemsTheSame(DisplayableItem, DisplayableItem)}
```

Comment:
This interface is a way to adapt the DiffUtils Callback to DisplayableItem. I took the documentation from the original platform's Callback and tried to adapt it. This part you refer to is in the original documentation and I found it useful when I was trying to understand how the DiffUtils works, that's why I decided to include it.
This is called by the DiffUtils (through the DiffUtilCallback class I implemented), I don't think there is a better place to include this documentation.